### PR TITLE
V0/jbn/vehicle data mock

### PR DESCRIPTION
**Pull request by ColdSwordX**
Fixed NameSpace Errors
And Relations

### CODE REVIEW

File: ITGuru.FourWheels/ITGuru.FourWheels.Data/Interfaces/IDataLayer.cs [hunk 1]
```diff
@@ -1,25 +1,21 @@
...
     {
         void GenerateList();
         List<Customer> GetAllCustomers();
+        List<Customer> GetAllDeletedCustomers();
         bool AddCustomer(Customer customer);
         bool UpdateCustomer(Customer customer);
         bool HardDeleteCustomer(Guid customerID);
         bool SoftDeleteCustomer(Guid customerID);
         List<Vehicle> GetAllVehicles();
+        List<Vehicle> GetAllDeletedVehicles();
```
Comment (ZhakalenDk) [inline ITGuru.FourWheels/ITGuru.FourWheels.Data/Interfaces/IDataLayer.cs]: `GetAllVehicles (Include deleted = false)`

**Review by ZhakalenDk — CHANGES_REQUESTED** [review]
Minor chnages are needed.

File: ITGuru.FourWheels/ITGuru.FourWheels.Data/Interfaces/IDataLayer.cs (hunk 1 above)
Comment (ColdSwordX) [reply ITGuru.FourWheels/ITGuru.FourWheels.Data/Interfaces/IDataLayer.cs]: Changes have been made       **GetAllVehicles() **and** GetAllCustomers()**     have been changed to    **GetAllVeicles(bool IncludeDeleted = false) **and** GetAllCustomers(bool IncludeDeleted = false)**

File: ITGuru.FourWheels/ITGuru.FourWheels.Data/Interfaces/IDataLayer.cs [hunk 2]
```diff
@@ -1,25 +1,19 @@
...
-using System.Text;
-using System.Threading.Tasks;
 
 namespace ITGuru.FourWheels.Data
 {
     public interface IDataLayer
     {
         void GenerateList();
-        List<Customer> GetAllCustomers();
+        List<Customer> GetAllCustomers(bool includeDeleted);
```
Comment (ZhakalenDk) [inline ITGuru.FourWheels/ITGuru.FourWheels.Data/Interfaces/IDataLayer.cs]: `includeDeleted = false`

File: ITGuru.FourWheels/ITGuru.FourWheels.Data/Interfaces/IDataLayer.cs [hunk 3]
```diff
@@ -1,25 +1,19 @@
...
         bool AddCustomer(Customer customer);
         bool UpdateCustomer(Customer customer);
         bool HardDeleteCustomer(Guid customerID);
         bool SoftDeleteCustomer(Guid customerID);
-        List<Vehicle> GetAllVehicles();
+        List<Vehicle> GetAllVehicles(bool includeDeleted);
```
Comment (ZhakalenDk) [inline ITGuru.FourWheels/ITGuru.FourWheels.Data/Interfaces/IDataLayer.cs]: `includeDeleted = false`

**Review by ZhakalenDk — CHANGES_REQUESTED** [review]
Still need some changes to be done

File: ITGuru.FourWheels/ITGuru.FourWheels.Data/Interfaces/IDataLayer.cs (hunk 3 above)
Comment (ColdSwordX) [reply ITGuru.FourWheels/ITGuru.FourWheels.Data/Interfaces/IDataLayer.cs]: Fixed

**Review by ZhakalenDk — APPROVED** [review]
Looks good. Make it happen!

**Pull request merged**